### PR TITLE
[Shadowlands][Mage] Conduits (Part 1)

### PR DESCRIPTION
**Pull request by Sharrq**
Added some Conduits and Legendaries for the mage specs. Towards #3675

Shared
- Added Grounding Surge conduit
- Added Diverted Energy conduit
- Added Ire of the Ascended conduit
- Adjusted Spellbook for Flow of Time conduit
- Adjusted Spellbook for Winters Protection conduit

Frost
- Added Ice Bite conduit
- Added Icy Propulsion conduit
- Added Shivering Core conduit
- Added Unrelenting Cold conduit

Fire
- Added Master Flame conduit

Misc
- Added Frost Mage to the CombatantInfo Faker
- Moved some talent modules to a talents folder
- Added Interrupt events to Events
- Legendary Spell and Bonus IDs

### CODE REVIEW

File: src/parser/mage/shared/modules/items/DivertedEnergy.tsx [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  overhealing = 0;
+
+  constructor(props: any) {
+    super(props);
+    this.active = this.selectedCombatant.hasConduitBySpellID(SPELLS.DIVERTED_ENERGY.id);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER).spell(SPELLS.DIVERTED_ENERGY_HEAL), this.onDivertedEnergyHeal);
+  }
+
+  onDivertedEnergyHeal(event: HealEvent) {
+    this.log(event);
```
Comment (Dambroda) [inline src/parser/mage/shared/modules/items/DivertedEnergy.tsx:24]: ;D

File: src/parser/mage/shared/modules/items/IreOfTheAscended.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+  }
+
+  onDamage(event: DamageEvent) {
+    if (!this.selectedCombatant.hasBuff(SPELLS.RADIANT_SPARK.id)) {
+      return;
+    }
+
+    const conduitRank = this.selectedCombatant.conduitRankBySpellID(SPELLS.IRE_OF_THE_ASCENDED.id);
+    this.bonusDamage += calculateEffectiveDamage(event, DAMAGE_BONUS[conduitRank]);
```
Comment (Dambroda) [inline src/parser/mage/shared/modules/items/IreOfTheAscended.tsx:46]: Just a comment for the future; once we can get some real testing done we should see if this applies to all damage or only to specific spells (like the class balance aura)

File: src/parser/mage/frost/modules/items/IcyPropulsion.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  cooldownReduction = 0;
+
+  constructor(props: any) {
+    super(props);
+    this.active = this.selectedCombatant.hasConduitBySpellID(SPELLS.ICY_PROPULSION.id);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.onDamage);
+  }
+
+  onDamage(event: DamageEvent) {
+    const conduitRank = this.selectedCombatant.conduitRankBySpellID(SPELLS.ICY_PROPULSION.id);
```
Comment (Dambroda) [inline src/parser/mage/frost/modules/items/IcyPropulsion.tsx:50]: Not a huge deal, but putting this calculation into the constructor and having a conduitRank module variable will save a lot of calls; throughout the PR.

File: src/parser/mage/shared/modules/items/IreOfTheAscended.tsx (hunk 2 above)
Comment (Sharrq) [reply src/parser/mage/shared/modules/items/IreOfTheAscended.tsx:46]: I believe radiant spark counts anything the player does. But that's something we can look into

File: src/parser/mage/shared/modules/items/DivertedEnergy.tsx (hunk 1 above)
Comment (Sharrq) [reply src/parser/mage/shared/modules/items/DivertedEnergy.tsx:24]: Oops, thought I removed all of them

**Pull request merged**